device/device.go

```diff
@@ -370,6 +374,7 @@ func (d *Device) Stop() error {
 		return fmt.Errorf("device: stop: %w", err)
 	}
 	d.streaming = false
+	d.cancelContext()
```

Comment:
I think the _only_ thing `Stop()` should do is cancel the context.

The context cancellation causes the goroutine's [for/select](https://github.com/vladimirvivien/go4vl/pull/36/files#diff-b4977e4fe2165fd3df54dc4180cbf6627972e511d3f6bd6e638caad460a39913R405) loop to break out and the other cleanup things should happen after this loop.

This ensures that the Stop sequence happens exactly the same whether the user calls `Stop()` or the parent context is cancelled.  Ie. the goroutine exits cleanly, and _then_ we clean up.

Optionally, if you want `Stop()` to be synchronous, there could be a `sync.WaitGroup` that waits for the stream loop goroutine to finish cleaning up.

Reply:
The context cancellation case in the stream loop goroutine would change to:

```go
case <-ctx.Done():
	break loop
```

- remove the call to `Stop()`
- make sure to add a `loop:` label before the `for` loop so we break out of the for loop and not just the select statement.